packages/desktop/src/main/main.ts

```diff
@@ -371,6 +371,21 @@ app.on('ready', async () => {
     closeBackendProcess()
   })
 
+  ipcMain.on('writeTempFile', (event, arg) => {
+    const temporaryFilesDirectory = path.join(appDataPath, 'temporaryFiles')
+    fs.mkdirSync(temporaryFilesDirectory, { recursive: true })
+    const filePath = `${path.join(temporaryFilesDirectory, arg.fileName)}`
+    const writeStream = fs.createWriteStream(filePath)
+    writeStream.write(arg.fileBuffer)
+    writeStream.end()
+
+    event.reply('writeTempFileReply', {
+      path: filePath,
+      id: arg.fileName.split(`.${arg.ext}`)[0],
```

Comment:
Does it mean that id will be the same for multiple files with the same name? It's a key and we use id for deleting file's preview

Reply:
No, every file will have a different id.